src/bscPlugin/validation/ScopeValidator.ts

```diff
@@ -90,16 +89,64 @@ export class ScopeValidator {
         outer:
         for (const info of expressionInfos) {
             const symbolTable = info.expression.getSymbolTable();
+            const fullName = info.parts.map(part => part.name.text).join('.');
             const firstPart = info.parts[0];
             const firstNamespacePart = info.parts[0].name.text;
             const firstNamespacePartLower = firstNamespacePart?.toLowerCase();
             //get the namespace container (accounting for namespace-relative as well)
             const namespaceContainer = scope.getNamespace(firstNamespacePartLower, info.enclosingNamespaceNameLower);
             let symbolType = SymbolTypeFlags.runtime;
+            let oppositeSymbolType = SymbolTypeFlags.typetime;
             const isUsedAsType = info.expression.findAncestor(isTypeExpression);
             if (isUsedAsType) {
                 // This is used in a TypeExpression - only look up types from SymbolTable
                 symbolType = SymbolTypeFlags.typetime;
+                oppositeSymbolType = SymbolTypeFlags.runtime;
+            }
+            let exprType = info.expression.getType(symbolType);
+            if (!exprType || !exprType.isResolvable()) {
+                if (info.expression.getType(oppositeSymbolType)?.isResolvable()) {
+                    const invalidlyUsedResolvedTypeName = info.expression.getType(oppositeSymbolType).toString();
+                    if (isUsedAsType) {
+
+                        this.addMultiScopeDiagnostic({
+                            ...DiagnosticMessages.itemCannotBeUsedAsType(fullName),
+                            range: info.expression.range,
+                            file: file
+                        }, 'When used in scope');
+                    } else {
+                        this.addMultiScopeDiagnostic({
+                            ...DiagnosticMessages.itemCannotBeUsedAsVariable(invalidlyUsedResolvedTypeName),
+                            range: info.expression.range,
+                            file: file
+                        }, 'When used in scope');
+                    }
+                    continue;
+                }
+
+                let fullErrorName = fullName;
+                let lastName = info.parts[info.parts.length - 1].name.text;
+                if (isDottedGetExpression(info.expression)) {
```

Comment:
gets the full name of the part of teh expression that cannot be found...

Eg.: `NS_ExistsA.NS_ExistsB.NS_DoesNotExist.MyClass` will say that ``NS_ExistsA.NS_ExistsB.NS_DoesNotExist` cannot be found